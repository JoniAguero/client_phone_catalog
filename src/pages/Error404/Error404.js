import React from "react";
import "./Error404.css";

export default function Error404() {
  return (
    <div>
      <h1>Error404...</h1>
    </div>
  );
}