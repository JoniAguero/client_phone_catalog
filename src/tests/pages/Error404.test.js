import React from 'react'
import { mount } from 'enzyme'
import { Provider } from 'react-redux'

import configureStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import '@testing-library/jest-dom'

import Error404 from '../../pages/Error404'

jest.mock('../../redux/actions/phonesActions', () => ({
  getPhones: jest.fn(),
}))

const middlewares = [thunk]
const mockStore = configureStore(middlewares)

const initialState = {
  phones: {
    list: [],
    phoneSelected: null
  },
  errors: {
    error: {}
  }
}

const store = mockStore(initialState)
store.dispatch = jest.fn()

const wrapper = mount(
  <Provider store={store}>
    <Error404 />
  </Provider>
)

describe('Error404 Page: ', () => {
  test('should be displayed correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
