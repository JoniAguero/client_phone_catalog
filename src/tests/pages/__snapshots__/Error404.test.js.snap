// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error404 Page:  should be displayed correctly 1`] = `
<div>
  <h1>
    Error404...
  </h1>
</div>
`;
